packages/appsync-modelgen-plugin/src/visitors/appsync-visitor.ts

```diff
@@ -693,6 +693,9 @@ export class AppSyncModelVisitor<
               isList: false,
               isNullable: field.isNullable,
             });
+          } else if (connectionInfo.targetName !== 'id') {
```

Comment:
nit: It is better to have this removing logic in a second loop. Otherwise it will introduce back an old issue. More info: https://github.com/aws-amplify/amplify-codegen/pull/323

Reply:
Please feel free to merge it if this is out of the scope of sev2.

Reply:
That is still an open issue. I would keep this change specific to addressing the studio issue, and have a separate release for the other.